core/vm/contracts_zkevm.go

```diff
@@ -299,6 +299,27 @@ func (c *bigModExp_zkevm) RequiredGas(input []byte) uint64 {
 	} else {
 		input = input[:0]
 	}
+
+	// Retrieve the operands and execute the exponentiation
+	var (
+		base       = new(big.Int).SetBytes(getData(input, 0, baseLen.Uint64()))
+		exp        = new(big.Int).SetBytes(getData(input, baseLen.Uint64(), expLen.Uint64()))
+		mod        = new(big.Int).SetBytes(getData(input, baseLen.Uint64()+expLen.Uint64(), modLen.Uint64()))
+		baseBitLen = base.BitLen()
+		expBitLen  = exp.BitLen()
+		modBitLen  = mod.BitLen()
+	)
+
```

Comment:
if modulo size is 0, the precompile returns 0 and it consumes GAS. This case is not handled.
if base size is 0 & modsize < 8192, it should also return 0 and consume GAS. This case is not handled.

Reply:
<img width="545" alt="image" src="https://github.com/user-attachments/assets/85a8167b-0197-4d7a-9dfc-728de811d56a">
These are the revert cases to return a 0 gas so that we don't consume any.  Anything else consumes gas as it would normally, unless I'm missing something?

Reply:
Just pinging here to revive the thread.  I believe if mod is 0 then gas is still charged normally and it will return 0 from the `Run` function.  Likewise if base is 0 and mod is < 8192 gas will be charged normally and 0 will be returned (or empty bytes).

Reply:
exactly. But in the implementation, if mod is 0 and any other param is above 8192, it will not consume GAS.
Same for base size is 0 & modsize < 8192

Reply:
Right, so the rules aren't combined?  They are independent, i.e. if mod = 0 then consume gas and just return 0 AND do not check for values being over 8192.

Reply:
yes

Reply:
same for base size is 0 & modsize < 8192